src/c-tips/c-tips.md

```diff
@@ -1,3 +1,209 @@
+  date or Git commit hash into your executable
+* generate linker scripts at build time depending on selected features or other
+  logic
+* change the Cargo build configuration
+* add extra static libraries to link against
+
+### Cross-Compiling
+
+Using Cargo for your build system also simplifies cross-compiling. In most
+cases it suffices to tell Cargo `--target thumbv6m-none-eabi` and find a
+suitable executable in `target/thumbv6m-none-eabi/debug/myapp`.
+
+For platforms not natively supported by Rust, you will need to build `libcore`
+for that target yourself. On such platforms, [Xargo] can be used as a stand-in
+for Cargo which automatically builds `libcore` for you.
+
+[Xargo]: https://github.com/japaric/xargo
+
+## Iterators vs Array Access
+
+In C you are probably used to accessing arrays directly by their index:
+
+```c
+int16_t arr[16];
+int i;
+for(i=0; i<sizeof(arr)/sizeof(arr[0]); i++) {
+    process(arr[i]);
+}
+```
+
+In Rust this is an anti-pattern: indexed access can be slower (as it needs to
+be bounds checked) and may prevent various compiler optimisations.
+
+Instead, use iterators:
+
+```rust
+let arr = [0u16; 16];
+for element in arr.iter() {
+    process(*arr);
+}
+```
+
+Iterators provide a powerful array of functionality you would have to implement
+manually in C, such as chaining, zipping, enumerating, finding the min or max,
+summing, and more. Iterator methods can also be chained, giving very readable
+data processing code.
+
+See the [Iterators in the Book] and [Iterator documentation] for more details.
+
+[Iterators in the Book]: https://doc.rust-lang.org/book/second-edition/ch13-02-iterators.html
+[Iterator method documentation]: https://doc.rust-lang.org/core/iter/trait.Iterator.html
+
+## References vs Pointers
+
+In Rust, pointers (called [_raw pointers_]) exist but are rarely used:
+dereferencing them is always considered `unsafe` as Rust cannot provide its
+usual guarantees about what might be behind the pointer.
+
+[_raw pointers_]: https://doc.rust-lang.org/book/second-edition/ch19-01-unsafe-rust.html#dereferencing-a-raw-pointer
+
+In most cases, we instead use _references_, indicated by the `&` symbol, or
+_mutable references_, indicated by `&mut`. References behave similarly to
+pointers, in that they can be dereferenced to access the underlying values, but
+they are a key part of Rust's ownership system: Rust will strictly enforce that
+you may only have one mutable reference _or_ multiple non-mutable references to
+the same value at any given time.
+
+In practice this means you have to be more careful about whether you need
+mutable access to data: where in C the default is mutable and you must be
+explicit about `const`, in Rust the opposite is true.
+
+## Other Resources
+
+* In this book: [A little C with your Rust](../interoperability/c-with-rust.md)
+* In this book: [A little Rust with your C](../interoperability/rust-with-c.md)
```

Comment:
**In this book** duplicated on following lines

Reply:
This is deliberate in so far as those two items are each in the book and the subsequent items are not. I think it would look OK rendered?

Reply:
I've updated the list to nest these two under a single "In this book", which looks better. Thanks for pointing it out.